src/main/java/com/alphawallet/attestation/NFTAttestation.java

```diff
@@ -1,22 +1,33 @@
 package com.alphawallet.attestation;
 
 import com.alphawallet.attestation.core.ASNEncodable;
+import com.alphawallet.attestation.core.SignatureUtility;
 import com.alphawallet.attestation.core.Validateable;
 import com.alphawallet.ethereum.ERC721Token;
-import org.bouncycastle.asn1.ASN1EncodableVector;
-import org.bouncycastle.asn1.ASN1Primitive;
-import org.bouncycastle.asn1.ASN1Sequence;
-import org.bouncycastle.asn1.DERSequence;
+import org.bouncycastle.asn1.*;
+import org.bouncycastle.crypto.params.AsymmetricKeyParameter;
+
+import java.io.IOException;
 
 public class NFTAttestation implements ASNEncodable, Validateable {
     private final SignedAttestation att;
     private final ASN1Sequence token;
 
-    public NFTAttestation(SignedAttestation att, ERC721Token nftToken) {
+    public NFTAttestation(SignedAttestation att, ERC721Token nftToken)
+    {
         this.att = att;
         this.token = new DERSequence(nftToken.getTokenVector());
     }
 
+    public NFTAttestation(byte[] derEncoding, AsymmetricKeyParameter signingPublicKey) throws IOException {
+        ASN1InputStream input = new ASN1InputStream(derEncoding);
+        ASN1Sequence asn1 = ASN1Sequence.getInstance(input.readObject());
+
+        ASN1Sequence attestationEnc = ASN1Sequence.getInstance(asn1.getObjectAt(0)); //root attestation, should be signed att
```

Comment:
This does not completely match the latest asn1 format [we have](https://github.com/TokenScript/attestation/blob/main/data-modules/output/asn/NFTAttestation.asn). However if what is in the code is what we want the it is super easy for me to change the asn1 description. But we should be sure this is the format we actually want